src/Company/Companies.ts

```diff
@@ -1,46 +1,30 @@
 // Constructs all CompanyPosition objects using the metadata in data/companypositions.ts
-import { companiesMetadata } from "./data/CompaniesMetadata";
-import { Company, IConstructorParams } from "./Company";
-import { Reviver } from "../utils/JSONReviver";
+import { getCompaniesMetadata } from "./data/CompaniesMetadata";
+import { Company } from "./Company";
+import { Reviver, assertLoadingType } from "../utils/JSONReviver";
+import { CompanyName } from "./Enums";
+import { createEnumKeyedRecord } from "../Types/Record";
+import { getEnumHelper } from "../utils/EnumHelper";
 
-export let Companies: Record<string, Company> = {};
-
-function addCompany(params: IConstructorParams): void {
-  if (Companies[params.name] != null) {
-    console.warn(`Duplicate Company Position being defined: ${params.name}`);
-  }
-  Companies[params.name] = new Company(params);
-}
-
-// Used to initialize new Company objects for the Companies map
-// Called when creating new game or after a prestige/reset
-export function initCompanies(): void {
-  // Save Old Company data for 'favor'
-  const oldCompanies = Companies;
-
-  // Re-construct all Companies
-  Companies = {};
-  companiesMetadata.forEach((e) => {
-    addCompany(e);
-  });
-
-  // Reset data
-  for (const companyName of Object.keys(Companies)) {
-    const company = Companies[companyName];
-    const oldCompany = oldCompanies[companyName];
-    if (!oldCompany) {
-      // New game, so no OldCompanies data
-      company.favor = 0;
-    } else {
-      company.favor = oldCompanies[companyName].favor;
-      if (isNaN(company.favor)) {
-        company.favor = 0;
-      }
-    }
-  }
-}
+export const Companies: Record<CompanyName, Company> = (() => {
+  const metadata = getCompaniesMetadata();
+  return createEnumKeyedRecord(CompanyName, (name) => new Company(metadata[name]));
+})();
 
 // Used to load Companies map from a save
 export function loadCompanies(saveString: string): void {
-  Companies = JSON.parse(saveString, Reviver);
+  const loadedCompanies = JSON.parse(saveString, Reviver) as unknown;
```

Comment:
@Snarling have you considered using [ts-reset](https://github.com/total-typescript/ts-reset)?

Reply:
Yeah most of the changes in that seem to be improvements on the ts defaults. Unfortunately I think there are several places we are currently abusing the default `any` type and would need significant rewrite to avoid needing an `as any`.

Although maybe it would be better to bring in ts-reset (at least a few select rules from it) and just add `as any` in the places where it was needed, which also would essentially flag the areas for us to address later. Given that the entire loading process is based on JSON.parse, we would basically need fully typesafe loading for all parts of the save file (currently I believe it is only Aliases, Factions, and now with this PR Companies that use typesafe loading instead of just loading the JSON and hoping it's the right structure).

Reply:
Yeah, I think it's better to add `as any` in places that are not ready yet than finding and marking all vulnerable places `as unknown`.

Reply:
Will look into adding this in a separate PR after this one.